main.py

```diff
@@ -118,44 +98,46 @@ def login(self):
             stored_password = self.database.get_password(user_id)
             if stored_password == hashed_password:
                 self.logged_in_user_id = user_id
-                print(f"Login successful! Welcome back, {self.database.get_username(user_id)}.")
-            else:
-                print("Incorrect password.")
-        else:
-            print("No account found with that email.")
+                log.info(
+                    "Login successful! Welcome back, %s.", 
+                    self.database.get_username(user_id)
+                )
+                return
+        # To protect user's emails and passwords,
+        # do not specify whether the password or email is incorrect.
+        log.info("Incorrect email or password.")
 
     def update_email(self):
         """Updates the email of the logged-in user."""
         if self.logged_in_user_id is None:
-            print("You must be logged in to update your email.")
+            log.info("You must be logged in to update your email.")
             return
 
         new_email = input("Enter your new email: ")
-        if self.verify_email(new_email) == "bad":
-            print("email not good")
+        if not self.verify_email(new_email):
             return
         self.database.update_email(self.logged_in_user_id, new_email)
-        print("Email updated successfully!")
-        
+        log.info("Email updated successfully!")
+
     def delete_user(self):
         """Deletes the logged in user"""
         if self.logged_in_user_id is None:
-            print("You must be logged in to update your email.")
+            log.info("You must be logged in to delete your account.")
             return
 
         self.database.delete_user(self.logged_in_user_id)
         self.logged_in_user_id = None
-        print("User deleted successfully")
+        log.info("User deleted successfully.")
 
     def start(self):
         """Main menu for the smart home app."""
         while True:
-            print("\nWelcome to the Smart Home App!")
-            print("1. Sign Up")
-            print("2. Log In")
-            print("3. Update Email")
-            print("4. Delete User")
-            print("5. Exit")
+            log.info("\nWelcome to the Smart Home App!")
+            log.info("1. Sign Up")
+            log.info("2. Log In")
+            log.info("3. Update Email")
+            log.info("4. Delete User")
+            log.info("5. Exit")
```

Comment:
```suggestion
            print("\nWelcome to the Smart Home App!")
            print("1. Sign Up")
            print("2. Log In")
            print("3. Update Email")
            print("4. Delete User")
            print("5. Exit")
```
`print` was correct here no? we wouldn't want to log this.

some of the other uses of log we maybe want to print instead as well, but this could be changed/decided in the future.